ts/utils/input.ts

```diff
@@ -67,26 +70,33 @@ export type CreditCardState = Readonly<{
 
 export type CreditCardStateKeys = keyof CreditCardState;
 
+// TODO: update this function including all the form errors when fp-ts will be updated.
+// Refers to https://dev.to/gcanti/getting-started-with-fp-ts-either-vs-validation-5eja
 /**
  * @param state A pending credit card state objects, containing the card's components.
- * @returns A fp-ts Option containing a valid CreditCard object, if possible
+ * @returns A fp-ts Either containing a valid CreditCard object if right or if it's left undefined if at least one field is none
+ * or the label of the first invalid field.
  */
 export function getCreditCardFromState(
   state: CreditCardState
-): Option<CreditCard> {
+): Either<string | undefined, CreditCard> {
   const { pan, expirationDate, securityCode, holder } = state;
   if (
     pan.isNone() ||
     expirationDate.isNone() ||
     securityCode.isNone() ||
     holder.isNone()
   ) {
-    return none;
+    return left(undefined);
```

Comment:
```suggestion
    return left("all field are required");
```
what do you think?
Since all fields are filled we have not feedback about the error (on accessibility)
In this way we are telling to the user that the button is disabled because the input is not complete

Reply:
Yes, also @thisisjp agree!
Refactored in [aa8546c](https://github.com/pagopa/io-app/pull/3020/commits/aa8546c71219348034ca0aab6ea6e86deb229f76)